### PR TITLE
add templates test verification testcase

### PR DESCRIPTION
**Pull request by sHaggYcaT**
**What this PR does / why we need it**:
This test case checks if common-templates.yaml is valid for UI or not.

### CODE REVIEW

File: tests/templates_verification_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	})
+
+	Context("Testing generated templates", func() {
+
+		// CNV-1069
+		It("Check if template valid for UI", func() {
+
+			// Getting common_templates
+			// TODO: replace downloading common-templates with getting common-templates from RPM
+			ct_yml_url := "https://github.com/kubevirt/common-templates/releases/download/v0.3.1/common-templates-v0.3.1.yaml"
```
Comment (MarSik) [inline tests/templates_verification_test.go]: Fixed version? What happens when we release a new one?

File: tests/templates_verification_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				By("Checking that template contains required lables")
+				Expect(search_for_pattern(yaml_segment, "\\s+os.template.cnv.io/[a-z0-9\\.]+:\\s\"true\"$")).To(BeTrue(), "Template should have os label")
+				Expect(search_for_pattern(yaml_segment, "\\s+workload.template.cnv.io/[a-z]+:\\s\"true\"$")).To(BeTrue(), "Template should have workload label")
+				Expect(search_for_pattern(yaml_segment, "\\s+flavor.template.cnv.io/[a-z]+:\\s\"true\"$")).To(BeTrue(), "Template should have flavor label")
+				Expect(search_for_pattern(yaml_segment, "\\s+template.cnv.io/type:\\s\"base\"$")).To(BeTrue(), "Template should have type base")
+
+			}
+		})
+	})
+})
```
Comment (MarSik) [inline tests/templates_verification_test.go]: Is this it? No checks for the presence of major operating systems for example?

**Comment by MarSik** [comment]
@cynepco3hahue @petrkotas can you please check this or add other QA reviewers? I am not sure who to add.

File: tests/templates_verification_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			common_templates_yaml := string(data)
+
+			for _, yaml_segment := range strings.Split(common_templates_yaml, "\n---\n") {
+
+				if !search_for_pattern(yaml_segment, "^Kind: Template$") {
+					continue
+				}
+
+				By("Checking that template contains required lables")
+				Expect(search_for_pattern(yaml_segment, "\\s+os.template.cnv.io/[a-z0-9\\.]+:\\s\"true\"$")).To(BeTrue(), "Template should have os label")
```
Comment (petrkotas) [inline tests/templates_verification_test.go]: How does the UI figures out that the template is valid? I would like to see it referenced here and than strictly follow the mechanics in the test.

**Comment by MarSik** [comment]
The UI doesn't do any validation. Openshift does when creating an instance. The labels and annotations are a convention that is described here https://github.com/kubevirt/common-templates/blob/master/templates/README.md

File: tests/templates_verification_test.go (hunk 3 above)
Comment (sHaggYcaT) [reply tests/templates_verification_test.go]: We have manual test plan. It was written based by recommendations from UI plan. I can share it you and can explain. Can we have some kind of video call or text chat or maybe email conversation?

**Comment by petrkotas** [comment]
I think we are good :)
--
s pozdravem,
Petr Kotas

**Comment by sHaggYcaT** [comment]
I've made rebase & fix (see description above)

File: tests/templates_verification_test.go (hunk 1 above)
Comment (sHaggYcaT) [reply tests/templates_verification_test.go]: I've implemented determination of the manifest's latest version (using github API)

File: tests/templates_verification_test.go (hunk 2 above)
Comment (sHaggYcaT) [reply tests/templates_verification_test.go]: I'm just following manual test plan that doesn't have any OS checks. @MarSik  What kind of check do you expect here?

File: tests/templates_verification_test.go (hunk 1 above)
Comment (sHaggYcaT) [reply tests/templates_verification_test.go]: @MarSik ^^

**Comment by MarSik** [comment]
@sHaggYcaT I do not have merge rights and the tests are not passing..

**Comment by MarSik** [comment]
ci please test

**Pull request merged**